src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntity_QuantumForceTransformer.java

```diff
@@ -0,0 +1,1160 @@
+                                            .atLeast(InputBus, InputHatch, Maintenance, Energy, ExoticEnergy)
+                                            .casingIndex(TAE.getIndexFromPage(0, 10))
+                                            .dot(4)
+                                            .buildAndChain(onElementPass(
+                                                    x -> ++x.mCasing, ofBlock(ModBlocks.blockCasings2Misc, 12)))))
+                    .addElement(
+                            'T',
+                            buildHatchAdder(GregtechMetaTileEntity_QuantumForceTransformer.class)
+                                    .atLeast(OutputBus, OutputHatch, Maintenance)
+                                    .casingIndex(TAE.getIndexFromPage(0, 10))
+                                    .dot(5)
+                                    .buildAndChain(
+                                            onElementPass(x -> ++x.mCasing, ofBlock(ModBlocks.blockCasings2Misc, 12))))
+                    .addElement(
+                            'Z',
+                            buildHatchAdder(GregtechMetaTileEntity_QuantumForceTransformer.class)
+                                    .hatchClass(GT_MetaTileEntity_Hatch_Input.class)
+                                    .adder(GregtechMetaTileEntity_QuantumForceTransformer::addNeptuniumHatch)
+                                    .casingIndex(TAE.getIndexFromPage(0, 10))
+                                    .dot(5)
+                                    .buildAndChain(
+                                            onElementPass(x -> ++x.mCasing, ofBlock(ModBlocks.blockCasings2Misc, 12))))
+                    .addElement(
+                            'X',
+                            buildHatchAdder(GregtechMetaTileEntity_QuantumForceTransformer.class)
+                                    .hatchClass(GT_MetaTileEntity_Hatch_Input.class)
+                                    .adder(GregtechMetaTileEntity_QuantumForceTransformer::addFermiumHatch)
+                                    .casingIndex(TAE.getIndexFromPage(0, 10))
+                                    .dot(5)
+                                    .buildAndChain(
+                                            onElementPass(x -> ++x.mCasing, ofBlock(ModBlocks.blockCasings2Misc, 12))))
+                    .build();
+        }
+        return this.STRUCTURE_DEFINITION;
+    }
+
+    public final boolean addCatalystHousing(IGregTechTileEntity aTileEntity, int aBaseCasingIndex) {
+        if (aTileEntity == null) {
+            return false;
+        } else {
+            IMetaTileEntity aMetaTileEntity = aTileEntity.getMetaTileEntity();
+            if (aMetaTileEntity instanceof GT_MetaTileEntity_Hatch_Catalysts) {
+                return addToMachineList(aTileEntity, aBaseCasingIndex);
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean addToMachineList(IGregTechTileEntity aTileEntity, int aBaseCasingIndex) {
```

Comment:
Maybe it would be a good idea to merge this function with `addCatalystHousing`, since they are almost identical.

Reply:
We are gonna be removing the catalyst housing itself as the catalyst won't be consumed. Pretty sure that was the plan